xarray/tests/test_units.py

```diff
@@ -0,0 +1,1515 @@
+import operator
+from distutils.version import LooseVersion
+
+import numpy as np
+import pytest
+
+import xarray as xr
+from xarray.core.npcompat import IS_NEP18_ACTIVE
+
+pint = pytest.importorskip("pint")
+DimensionalityError = pint.errors.DimensionalityError
+
+
+pytestmark = [
+    pytest.mark.skipif(
+        not IS_NEP18_ACTIVE, reason="NUMPY_EXPERIMENTAL_ARRAY_FUNCTION is not enabled"
+    ),
+    # pytest.mark.filterwarnings("ignore:::pint[.*]"),
+]
+
+
+def use_pint_version_or_xfail(*, version, reason):
+    return pytest.mark.xfail(LooseVersion(pint.__version__) < version, reason=reason)
+
+
+unit_registry = pint.UnitRegistry()
+require_pint_array_function = pytest.mark.skipif(
+    not hasattr(unit_registry.Quantity, "__array_function__"),
+    reason="pint does not implement __array_function__ yet",
+)
+
+try:
+    from pint.quantity import BaseQuantity
+except ImportError:
+    BaseQuantity = unit_registry.Quantity
```

Comment:
This is not merged into master, and I'm arguing on the pint board that it is likely a bad design.
https://github.com/hgrecco/pint/pull/880 is an alternative, cleaner approach.
For compatibility and until either PR is merged and a release is published off it, I think this whole module should just use
```python
Quantity = pint.UnitRegistry().Quantity
```

Reply:
I have been developing all this against hgrecco/pint#764, so there may be more references to it. That said, replacing with
```python
Quantity = unit_registry.Quantity
```
seems to work